tests/e2e/README.md

```diff
@@ -1,117 +1,144 @@
 ```
-source istio.VERSION
-# Each time pilot SHA changes, get the matching istioctl binary
-# for instance on a mac:
-curl $ISTIOCTL_URL/istioctl-osx > ~/istioctl-osx ; chmod 755 ~/istioctl-osx
-# Each time the test code changes, rebuild the test, for instance:
-bazel build //tests/e2e/tests/simple:go_default_test
-# First time you want to run: deploy in namespace e2e and leave it running:
-./bazel-bin/tests/e2e/tests/simple/go_default_test -alsologtostderr -test.v -v 2  --skip_cleanup --namespace=e2e -istioctl ~/istioctl-osx --auth_enable
-# Subsequent runs if only the TestSimpleIngress (for instance) changes:
-./bazel-bin/tests/e2e/tests/simple/go_default_test -alsologtostderr -test.v -v 2  --skip_setup --skip_cleanup --namespace=e2e -istioctl ~/istioctl-osx --auth_enable --test.run TestSimpleIngress
+export HUB="gcr.io/istio-testing"
+export GIT_SHA="<sha copied from the logs of istio-presubmit.sh>"
 ```
 
+## Step 5: Run
+```
+./tests/e2e.sh --mixer_tag "${GIT_SHA}"  --mixer_hub "${HUB}"  --pilot_tag "${GIT_SHA}"  --pilot_hub "${HUB}"  \
+--ca_tag "${GIT_SHA}"  --ca_hub "${HUB}"  --istioctl_url "https://storage.googleapis.com/istio-artifacts/pilot/${GIT_SHA}/artifacts/istioctl"   \
+--skip_cleanup
+```
+Tests are driven by the [e2e.sh](../e2e.sh) script. Each test has its own directory and can be run independently as a
+go_test target. The script has a number of options:
+
+* `--skip_cleanup` - to skip cleanup steps
+* `--namespace <namespace>` : If you don't specify `namespace`, a random namespace is generated for each test.
+* `--verbose <debug level noise from proxies>`
+* `--istioctl_url <location of istioctl>`
+* `--use_local_cluster`
+* `--parallel` - to run tests in parallel
+* `--auth_enable` - if you want to include auth
+* `--cluster_wide` - if you want to run the cluster wide installation and tests
+* `--use_initializer` - if you want to do transparent sidecar injection
+* `--mixer_hub <mixer image hub>`
+* `--mixer_tag <mixer image tag>`
+* `--pilot_hub <pilot image hub>`
+* `--pilot_tag <pilot image tag>`
+* `--ca_hub <CA image hub>`
+* `--ca_tag <CA image tag>`
+
+## Examples
 
-## Access to logs and temp files from Jenkins
-
-If tests ran in presubmit on Jenkins, you can easily access to logs and temp files. Go to your pr page on Jenkins console "https://testing.istio.io/job/istio/job/presubmit/<pr_number>", click "artifacts.html", and it will lead you to tests records.
+From the repo checkout root directory
 
-## demo_test.go
+* Running single category of test (bookinfo, mixer, simple) and also skip cleanup. Add "-s" flag.
+```
+./tests/e2e.sh --mixer_tag "${GIT_SHA}"  -s bookinfo --mixer_hub "${HUB}"  --pilot_tag "${GIT_SHA}"  \
+--pilot_hub "${HUB}"  --ca_tag "${GIT_SHA}"  --ca_hub "${HUB}"  \
+--istioctl_url "https://storage.googleapis.com/istio-artifacts/pilot/${GIT_SHA}/artifacts/istioctl"   --skip_cleanup
+```
+* TODO: THIS IS OUTDATED. If you want to run one specific test, you can do:
```

Comment:
If it's outdated, should we remove it or fix it?

Reply:
I don't know how to make it work.. would need help. I have created https://github.com/istio/istio/issues/2036 for someone to update.

Reply:
it's a very important use case: rerun 1 test multiple times while leaving the namespace intact in case of failure (to investigate a failure)